packages/Kokkos/test/tstDetailsTreeConstruction.cpp

```diff
@@ -106,46 +117,61 @@ TEUCHOS_UNIT_TEST( DetailsBVH, number_of_leading_zero_bits )
     TEST_EQUALITY( dtk::countLeadingZeros( 4 ^ 3 ), 29 );
 }
 
-TEUCHOS_UNIT_TEST( DetailsBVH, common_prefix )
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DetailsBVH, common_prefix, NO )
 {
+    using DeviceType = typename NO::device_type;
+    int const n = 13;
     // NOTE: Morton codes below are **not** unique
-    std::vector<unsigned int> const fi = {
-        0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144,
-    };
-    int const n = fi.size();
-    TEST_EQUALITY( dtk::commonPrefix( fi.data(), n, 0, 0 ), 32 + 32 );
-    TEST_EQUALITY( dtk::commonPrefix( fi.data(), n, 0, 1 ), 31 );
-    TEST_EQUALITY( dtk::commonPrefix( fi.data(), n, 1, 0 ), 31 );
+    Kokkos::View<unsigned int *, DeviceType> fi( "fi", n );
+    fi[0] = 0;
+    fi[1] = 1;
+    fi[2] = 1;
+    fi[3] = 2;
+    fi[4] = 3;
+    fi[5] = 5;
+    fi[6] = 8;
+    fi[7] = 13;
+    fi[8] = 21;
+    fi[9] = 34;
+    fi[10] = 55;
+    fi[11] = 89;
+    fi[12] = 144;
+
+    DataTransferKit::TreeConstruction<NO> tc;
+    TEST_EQUALITY( tc.commonPrefix( fi, n, 0, 0 ), 32 + 32 );
```

Comment:
No we don't need one. I first created the structure and afterwards I made all the functions static. So we can choose if we want to use the structure or not.